### ISSUE TITLE
bug: In spec tests : ReferenceError: ShadowRoot is not defined

### ISSUE THREAD

**Issue by ryuran** [ionic-team/stencil#3260]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.14.0

### Current Behavior

`this.el.getRootNode() instanceof ShadowRoot`
I try to know if my custom component is in a shadowDOM.
It works well but not in specs test.
I have this error `ReferenceError: ShadowRoot is not defined`

### Expected Behavior

No error in test when we use `ShadowRoot`

### Steps to Reproduce

Try to test if the host element is in a shadowDOM context
`this.el.getRootNode() instanceof ShadowRoot`

Then run unit tests.

### Code Reproduction URL

https://github.com/ryuran/stencil-bugs

### Additional Information

- `npm i`
- `npm test`

### Workaround
Mock:
```
global.DocumentFragment = class DocumentFragment extends Node {};
global.ShadowRoot = class ShadowRoot extends DocumentFragment {};
```

**Comment by rwaskiewicz** [comment]
Thanks! I've verified this is an issue and have labeled this to be ingested into our team backlog!

**Comment by reessm01** [comment]
I just wanted to chime in that I was enduring shadowRoot not defined as well (no reference or clue or even what is different as the error information isn't helpful at all), however the work around provided here fixed it.

**Comment by tanner-reits** [comment]
A fix for this was included in today's [v4.19.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.19.0)!
